WordPress/gutenberg
# Updates the behavior of the top toolbar fixed setting

## What?


Closes #40450
Updates the behavior of the top toolbar fixed setting.

## Why?


## How?


- wraps the edit post header in a slot fill provider
- implements a temporary behavior that shows document tools when no block is selected
- updates the block tools and parent selector to appear properly in the new place
- implements a utility component that shows the document tools when they receive focus via toolbar

## Testing Instructions





1. Enable the Top Toolbar setting from the top right dot menu
2. Select a block
3. Notice the block tools (toolbar) replaces the document tools (toolbar)

### Testing Instructions for Keyboard


N/A

## Screenshots or screencast 


https://user-images.githubusercontent.com/107534/225633841-3fdd191f-795e-424c-aeca-762443b74c6d.mp4
